dipy/tracking/local/localtracking.py

```diff
@@ -93,12 +100,22 @@ def _generate_streamlines(self):
             directions = dg.initial_direction(s)
             directions = directions[:max_cross]
             for first_step in directions:
-                stepsF = local_tracker(dg, tc, s, first_step, vs, F, ss, fixed)
-                if stepsF < 0:
+                stepsF, tissue_class = local_tracker(dg, tc, s, first_step,
+                                                     vs, F, ss, fixed)
+                # enum TissueClass (tissue_classifier.pyd) is not accessible
+                # from here. To be changed when minimal cyhton version > 0.21.
+                # cython 0.21 - cpdef enum to export values into Python-level namespace
+                # https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+                if not (self.return_all or
+                        tissue_class == 2 or  # ENDPOINT
+                        tissue_class == -1):  # OUTSIDEIMAGE
```

Comment:
I ran into something and I'm not sure the best way to handle it. It's possible for `tissue_class` to be TRACKPOINT, this happens when `dg.get_direction` cannot return a direction for some point (ie the peak is > max_angle degrees from the incoming direction). Do we want treat these types of tracks the same as INVALIDPOINT (only return them if return_all is set to True)?